src/views/SceneView.js

```diff
@@ -29,8 +29,30 @@ export default class SceneView extends PureComponent<void, Props, void> {
     };
   }
 
+  componentWillMount() {
+    if (this.props.screenProps !== undefined && this.props.hideScreenPropsWarning) {
```

Comment:
Isnt this logic inverted? I think you want `&& !this.props.hideScreenPropsWarning`

I'm not super happy about this part. I like to warn people, but encouraging them to dismiss the warning in this way feels odd. I would be happier with a thing like this:

```
// This would dismiss all warnings about API changes prior to this version.
ReactNavigation.setWarningLevel('1.0.0-beta.2');
```